src/swift/Source.swift

```diff
@@ -144,132 +150,151 @@ public extension DispatchSource {
 			.delete, .write, .extend, .attrib, .link, .rename, .revoke]
 	}
 
+#if HAVE_MACH
 	public class func machSend(port: mach_port_t, eventMask: MachSendEvent, queue: DispatchQueue? = nil) -> DispatchSourceMachSend {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_mach_send(), UInt(port), eventMask.rawValue, queue) as DispatchSourceMachSend
+		let source = dispatch_source_create(_swift_dispatch_source_type_mach_send(), UInt(port), eventMask.rawValue, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceMachSend
 	}
+#endif
 
+#if HAVE_MACH
 	public class func machReceive(port: mach_port_t, queue: DispatchQueue? = nil) -> DispatchSourceMachReceive {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_mach_recv(), UInt(port), 0, queue) as DispatchSourceMachReceive
+		let source = dispatch_source_create(_swift_dispatch_source_type_mach_recv(), UInt(port), 0, queue?.__wrapped)
+		return DispatchSource(source) as DispatchSourceMachReceive
 	}
+#endif
 
+#if HAVE_MACH
 	public class func memoryPressure(eventMask: MemoryPressureEvent, queue: DispatchQueue? = nil) -> DispatchSourceMemoryPressure {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_memorypressure(), 0, eventMask.rawValue, queue) as DispatchSourceMemoryPressure
+		let source = dispatch_source_create(_swift_dispatch_source_type_memorypressure(), 0, eventMask.rawValue, queue.__wrapped)
+		return DispatchSourceMemoryPressure(source)
 	}
+#endif
 
+#if HAVE_MACH
```

Comment:
This is unfortunate that we can't have processes in dispatch, what is the limiter here? It would be super nice to have for implementing parts of Foundation